internal/component/artifactory_release_source.go

```diff
@@ -44,6 +44,7 @@ type ArtifactoryFolderInfo struct {
 		URI    string `json:"uri"`
 		Folder bool   `json:"folder"`
 	} `json:"children"`
+	Path string `json:"path"`
```

Comment:
I introduced path so we ensure that the entire path is being matched on. For example
```json
{
  "repo" : "a-repo-dev-local",
  "path" : "/compiled-releases/release",
  "created" : "2024-04-23T22:44:56.231Z",
  "createdBy" : "some-user",
  "lastModified" : "2024-04-23T22:44:56.231Z",
  "modifiedBy" : "some-user",
  "lastUpdated" : "2024-04-23T22:44:56.231Z",
  "children" : [ {
    "uri" : "/release-0.102.0-ubuntu-jammy-1.406.tgz",
    "folder" : false
  }, {
    "uri" : "/release-0.103.0-ubuntu-jammy-1.406.tgz",
    "folder" : false
  },
...
}
```

should not only look at the children's uris but also the path `/compiled-releases/release`, which must match as well. Including if it includes stemcell or version info